core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -701,6 +726,51 @@ private RecordCursorFactory generateFilter0(RecordCursorFactory factory, QueryMo
                 f.close();
             }
         }
+
+        boolean useJit = executionContext.getJitMode() != SqlJitMode.JIT_MODE_DISABLED;
+        if (useJit) {
+            final boolean optimize = factory.supportPageFrameCursor() && JitUtil.isJitSupported();
+            if (optimize) {
+                try (TableReader reader = engine.getReader(
```

Comment:
Use of `TableReader` here is dubious. Filter is allowed to be several layers removed from table. This code should only rely on `factory` and what it provides.

Reply:
Makes sense. Switched to `RecordCursorFactory#getPageFrameCursor()` in 534f481e3e8c39978522842849b1e1f52b7571e2. WDYT?